app/models/user.rb

```diff
@@ -104,10 +104,8 @@ def format_phone_number
   end
 
   def validate_phone_number
-    if phone.blank?
-      errors.add :phone, :blank
-    elsif !MultiFactorAuth.valid?(phone)
-      errors.add :phone, :invalid
+    if phone.present?
+      errors.add :phone, :invalid unless MultiFactorAuth.valid?(phone)
     end
   end
```

Comment:
Could this just be
```ruby
def validate_phone_number
  errors.add(:phone, :invalid) if MultiFactorAuth.invalid?(phone)
end
```
calling it as
```ruby
validate :validate_phone_number, if: phone.present?
```
(where `invalid?` just negates `valid?`)?